src/protocols/ethereum/codegen/abi.js

```diff
@@ -208,6 +225,54 @@ module.exports = class AbiCodeGenerator {
             `return new ${paramsClassName}(this)`,
           ),
         )
+
+        // Fixture generation
+        try {
+          const args = yaml.parse(fs.readFileSync('./fixtures.yaml', 'utf8'))
+          const blockNumber = args['blockNumber']
+          const contractAddr = args['contractAddr']
+          const topic0 = args['topic0']
+          const apiKey = args['apiKey']
+          const url = `https://api.etherscan.io/api?module=logs&action=getLogs&fromBlock=${blockNumber}&toBlock=${blockNumber}&address=${contractAddr}&${topic0}=topic0&apikey=${apiKey}`;
+
+          let resp = request("GET", url)
+          let body = JSON.parse(resp.getBody("utf8"))
+          if (body.status === '0') {
+            throw new Error(body.result)
+          }
+
+          let res = web3.eth.abi.decodeLog(
```

Comment:
I was wondering too if shouldn't we use the `decoding` library directly instead of importing the whole `web3` here.

It actually uses a smaller library under the hood.